crates/roc_std/src/roc_list.rs

```diff
@@ -489,62 +501,6 @@ impl<T, const SIZE: usize> From<[T; SIZE]> for RocList<T> {
     }
 }
 
-impl<'a, T> IntoIterator for &'a RocList<T> {
-    type Item = &'a T;
-    type IntoIter = core::slice::Iter<'a, T>;
-
-    fn into_iter(self) -> Self::IntoIter {
-        self.as_slice().iter()
-    }
-}
```

Comment:
Oops, looks like we went a bit too far - this `InterIterator` for `&'a RocList` is totally fine, just not the one beloew for owned `RocList`! 😅